### PR TITLE
fix: improve documentation

### PR DESCRIPTION
**Pull request by metacosm**
**What type of PR is this?**
/kind documentation
[skip ci]

**What does does this PR do / why we need it**:
Improves the documentation regarding using devfiles.

### CODE REVIEW

File: docs/public/deploying-a-devfile-using-odo.adoc [hunk 1]
```diff
@@ -8,21 +8,21 @@ With a devfile you can describe:
...
 * The source code being used
 * Development components such as IDE tools (VSCode) and application runtimes (Yarn / NPM)
-* A list of pre-defined commands that can be ran
+* A list of pre-defined commands that can be run
 * Projects to initially clone
 
 Odo takes this devfile and transforms it into a workspace of multiple containers running on OpenShift, Kubernetes or Docker.
 
-Devfile's are YAML files with a defined structure, take a look at the general https://github.com/redhat-developer/devfile/blob/master/docs/devfile.md[schema] of devfile.
+Devfiles are YAML files with a defined https://github.com/redhat-developer/devfile/blob/master/docs/devfile.md[schema].
```
Comment (mik-dass) [inline docs/public/deploying-a-devfile-using-odo.adoc:16]: `with a defined` what?

File: docs/public/deploying-a-devfile-using-odo.adoc [hunk 2]
```diff
@@ -8,21 +8,21 @@ With a devfile you can describe:
...
 
 == Odo and devfile
 
-When deploying a devfile using odo, odo will automatically look at the default https://github.com/elsony/devfile-registry[devfile] https://github.com/eclipse/che-devfile-registry/[registries]. Interacting with the devfile registries allows a user to pull a standard `devfile.yaml` and begin development immediately.
+Odo can now create components from devfiles as recorded in registries. Odo consults automatically the default https://github.com/elsony/devfile-registry[devfile] https://github.com/eclipse/che-devfile-registry/[registries] but users can also add their own registries. Devfiles contribute new component types that users can pull to begin development immediately.
```
Comment (mik-dass) [inline docs/public/deploying-a-devfile-using-odo.adoc:20]: `ODO automatically consults`

**Comment by mik-dass** [comment]
@boczkowska Can you please have a look at the PR?

**Comment by metacosm** [comment]
What's needed to merge that PR?

**Pull request merged**